src/Tizen.Wearable.CircularUI.Forms.Renderer/Shell/NavigationDrawer.cs

```diff
@@ -0,0 +1,305 @@
+
+            return await tcs.Task;
+        }
+
+        protected override IntPtr CreateHandle(EvasObject parent)
+        {
+            _gestureBox = new Box(parent);
+            return _gestureBox.Handle;
+        }
+
+        void Initialize()
+        {
+            AlignmentX = -1;
+            AlignmentY = -1;
+            WeightX = 1;
+            WeightY = 1;
+
+            _gestureBox.RepeatEvents = true;
+
+            _outterBox = new Box(this);
+            _outterBox.Show();
+
+            _contentBox = new Box(this);
+            _contentBox.PropagateEvents = false;
+
+            _label = new ElmSharp.Label(_outterBox)
+            {
+                BackgroundColor = EColor.Pink,
+            };
+
+            _contentBox.Show();
+            _label.Show();
+
+            _outterBox.PackEnd(_contentBox);
+            _outterBox.PackEnd(_label);
+
+            _label.StackAbove(_contentBox);
+
+            _gestureLayer = new GestureLayer(_gestureBox);
+            _gestureLayer.Attach(_gestureBox);
+
+            _gestureLayer.SetMomentumCallback(GestureLayer.GestureState.Start, OnStart);
+            _gestureLayer.SetMomentumCallback(GestureLayer.GestureState.End, OnEnd);
+            _gestureLayer.SetMomentumCallback(GestureLayer.GestureState.Abort, OnEnd);
+
+            _dragGestureLayer = new GestureLayer(_outterBox);
+            _dragGestureLayer.Attach(_outterBox);
+
+            _dragGestureLayer.SetMomentumCallback(GestureLayer.GestureState.Move, OnDragged);
+            _dragGestureLayer.SetMomentumCallback(GestureLayer.GestureState.End, OnDragEnd);
+            _dragGestureLayer.SetMomentumCallback(GestureLayer.GestureState.Abort, OnDragEnd);
+
+            _gestureBox.SetLayoutCallback(OnLayout);
+            _outterBox.SetLayoutCallback(OnContentLayout);
+
+            RotaryEventManager.Rotated += OnRotate;
+        }
+
+        async void OnRotate(EventArgs args)
+        {
+            if(_fadeInCancelTokenSource != null)
+            {
+                _fadeInCancelTokenSource.Cancel();
+            }
+
+            _fadeInCancelTokenSource = new CancellationTokenSource();
+            var token = _fadeInCancelTokenSource.Token;
+
+            if (!_isOpen)
+            {
+                await HideAsync(_outterBox);
+
+                _ = ShowAsync(_outterBox, cancelltaionToken:token);
+            }
+        }
+
+        void OnLayout()
+        {
+            var geometry = _parent.Geometry;
+            geometry.Y = (_isOpen) ? 0 : (geometry.Height - MinimumDrawerHeight);
+
+            _gestureBox.Geometry = _parent.Geometry;
+            _outterBox.Geometry = geometry;
+        }
+
+        void OnContentLayout()
+        {
+            var geometry = _outterBox.Geometry;
+            geometry.Height = MinimumDrawerHeight;
+
+            _contentBox.Geometry = _outterBox.Geometry;
+            _label.Geometry = geometry;
+        }
+
+        void OnStart(GestureLayer.MomentumData moment)
```

Comment:
How about change name to `OnDragStarted`?

Reply:
Fixed.